From 8e87bb891452150e368a19624785893016218fd9 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:16:08 -0400
Subject: [alsa] hda: Avoid unnecessary verbs write in snd_hda_activate_path()

Message-id: <140972823573-68-patchset-jkysela@redhat.com>
Patchwork-id: 90768
O-Subject: [RHEL 7.1 PATCH 067/442] ALSA: hda - Avoid unnecessary verbs write in snd_hda_activate_path()
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

... by using snd_Hda_codec_update_cache() instead of *_write_cache().
Since all path elements should have been updated by this function,
we are safe to assume that the cache contents are consistent.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 8f0972dfa9b7378aaf14eebf2454abd93a6033ad)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/hda_generic.c b/sound/pci/hda/hda_generic.c
index 8321a97..437ef13 100644
--- a/sound/pci/hda/hda_generic.c
+++ b/sound/pci/hda/hda_generic.c
@@ -762,7 +762,7 @@ void snd_hda_activate_path(struct hda_codec *codec, struct nid_path *path,
           AC_PWRST_D0);
   }
   if (enable && path->multi[i])
-   snd_hda_codec_write_cache(codec, nid, 0,
+   snd_hda_codec_update_cache(codec, nid, 0,
          AC_VERB_SET_CONNECT_SEL,
          path->idx[i]);
   if (has_amp_in(codec, path, i))
-- 
1.7.1